### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by Boreeas**
* Better variable naming
* Deduplicate CubePrimer.DEFAULT_STATE
* Move some static methods to the bottom
* Fix a bug in the VanillaCompatibilityGenerator's scan heuristic
* Also make runtime for the scan linear instead of quadratic
* Also make the code a bit clearer

### CODE REVIEW

File: src/main/java/cubicchunks/worldgen/generator/vanilla/VanillaCompatibilityGenerator.java [hunk 1]
```diff
@@ -70,31 +69,26 @@ public VanillaCompatibilityGenerator(IChunkGenerator vanilla, ICubicWorld world)
...
-			for (int z = 0; z < Cube.SIZE; z++) {
-				IBlockState blockState = lastChunk.getBlockState(x, 0, z);
-				int count = blockHistogram.getOrDefault(blockState, 0);
-				blockHistogram.put(blockState, count + 1);
+		// We scan three layers, because the bottom-most layer is usually fully bedrock
+		for (int y = 0; y < 3; y++) {
+			for (int x = 0; x < Cube.SIZE; x++) {
+				for (int z = 0; z < Cube.SIZE; z++) {
+					IBlockState blockState = lastChunk.getBlockState(x, 0, z);
+					if (blockState.getBlock() == Blocks.BEDROCK) continue; // Never use bedrock for world extension
```
Comment (Barteks2x) [inline src/main/java/cubicchunks/worldgen/generator/vanilla/VanillaCompatibilityGenerator.java]: I'm not sure if this formatting is allowed

Comment (Boreeas) [inline src/main/java/cubicchunks/worldgen/generator/vanilla/VanillaCompatibilityGenerator.java]: I reformatted the whole file in that commit, so it should be legal

Comment (Boreeas) [inline src/main/java/cubicchunks/worldgen/generator/vanilla/VanillaCompatibilityGenerator.java]: That being said, if you'd prefer to have it on top I can do that

**Comment by xcube16** [comment]
Can you squash some of the commits?

**Comment by Barteks2x** [comment]
Only the few most recent ones with fixes, not everything.

**Comment by xcube16** [comment]
maybe 015bbfe could be squashed into 8d57d3f

**Comment by Boreeas** [comment]
Rebased: squashed 015bbfe into 8d57d3f and adfe5ed/ddb4afb into ded4145

**Pull request merged**